juju/juju
# [JUJU-3459] Move resource repository to a shareable location.

Anticpating use of ResolveResources by both the resource and application facades. It could be moved to apiserver common, however there is already a charm related repository in core/charm. Avoiding have 2 different things called a charm repository.

## QA steps

*Commands to run to verify that the change works.*

```sh
$ juju deploy juju-qa-test
Located charm "juju-qa-test" in charm-hub, revision 19
Deploying "four" from charm-hub charm "juju-qa-test", revision 19 in channel stable on ubuntu@20.04/stable
$ juju resources juju-qa-test
Resource  Supplied by  Revision
foo-file  store        2

# once the unit is active/idle
$ juju config juju-qa-test foo-file=true
# observer change in unit message during juju status
$ juju status
...
juju-qa-test/0*  active    idle        0        10.74.147.98           resource line one: testing two.

# deploy with a specific resource revision
$ juju deploy juju-qa-test --resource foo-file=4 four
Located charm "juju-qa-test" in charm-hub, revision 19
Deploying "four" from charm-hub charm "juju-qa-test", revision 19 in channel stable on ubuntu@20.04/stable
$ juju resources four
Resource  Supplied by  Revision
foo-file  store        4

# once the unit is active/idle
$ juju config four foo-file=true
# observer change in unit message during juju status
$ juju status
...
four/0*          active    idle   1        10.74.147.103          resource line one: testing four.
```
